### PR TITLE
Pass the body to the stream handler, not the whole frame

### PR DESCRIPTION
**Pull request by stillinbeta**

### CODE REVIEW

File: src/http2_stream.erl [hunk 1]
```diff
@@ -330,14 +330,14 @@ open({recv_frame,
       {#frame_header{
           flags=Flags,
           type=?DATA
-         },_}=F},
+         },{data, Payload}}=F},
```
Comment (joedevivo) [inline src/http2_stream.erl]: This is too sneaky. please do something like

```erlang
#data{data=Payload}
```

File: src/http2_stream.erl [hunk 2]
```diff
@@ -350,14 +350,14 @@ open({recv_frame,
       {#frame_header{
               flags=Flags,
               type=?DATA
-         }, _Payload}=F},
+         }, {data, Payload}}=F},
```
Comment (joedevivo) [inline src/http2_stream.erl]: Same here

File: src/http2_connection.erl [hunk 3]
```diff
@@ -444,7 +444,7 @@ route_frame({H, Payload},
                         0;
                     NewIWS ->
                         lager:debug("old IWS: ~p new IWS: ~p", [OldIWS, NewIWS]),
-                        OldIWS - NewIWS
+                        NewIWS - OldIWS
```
Comment (stillinbeta) [inline src/http2_connection.erl]: maybe this should be merged elsewhere we can test it properly?

Comment (joedevivo) [inline src/http2_connection.erl]: The ol' switcheroo!

Comment (joedevivo) [inline src/http2_connection.erl]: For better or worse, it's a fix so I'm likely to include it here. I'll open an issue to deal with it.

**Pull request merged**